### PR TITLE
Improve doc strings for distributions

### PR DESCRIPTION
**Pull request by tblazina**
This is my first PR to the project, so I thought I'd start with something simple. This would address #2992 and is following the format established in #3022. I am happy to continue and do the rest of the distributions, but am wondering if the format is correct and that things would stay consistent where, for example, the `random()` and `logp()` methods for all distribution classes have the same doc string except for changes in the distribution it is referring too (e.g. "Draw random values from SomeRandom Distribution."), or if we would want to be a bit more explicit in the doc strings if there is specific logic in the method (e.g. see `random()` method for `MatrixNormal()` distribution in `multivariate.py`)? 

Also I noticed that there doesn't seem to be logic for ordering of class methods in the distributions, sometimes private methods appear before public methods and sometimes not. It's not really that important but i'd be also happy to organize them if there was a preferred style.

### CODE REVIEW

File: pymc3/distributions/discrete.py [hunk 1]
```diff
@@ -66,12 +66,54 @@ def __init__(self, n, p, *args, **kwargs):
...
+        ----------
+        value : numeric
+            Value(s) for which log-probability is calculated. If the log probabilities for multiple
+            values are desired the values must be provided in a numpy array or theano tensor
+
+        Returns
+        -------
+        TensorVariable
+        """
+        """
```
Comment (canyon289) [inline pymc3/distributions/discrete.py]: Is this docstring duplicated?

Comment (tblazina) [reply pymc3/distributions/discrete.py]: Whoops! good catch

**Comment by canyon289** [comment]
Thank you for the contribution, and linking to relevant issue tickets, both are helpful. 

I find the docstrings to be helpful and from my understanding they are correct, although I'll let one of the more seasoned folks weigh in as well. As a first pass just having doc strings is an improvement.

As for the ordering of public and private classes I'm assuming there is not preferred style due to the random ordering :) but again will let one of the more seasoned folks weigh in.

Thanks again for taking the time to submit a PR!

**Review by twiecki — APPROVED** [review]

**Comment by canyon289** [comment]
Thanks @tblazina

**Pull request merged**